Lib/test/test_socket.py

```diff
@@ -1785,6 +1782,44 @@ def test_socket_fileno(self):
             s.bind(os.path.join(tmpdir, 'socket'))
             self._test_socket_fileno(s, socket.AF_UNIX, socket.SOCK_STREAM)
 
+    def test_socket_fileno_rejects_float(self):
+        with self.assertRaises(TypeError):
```

Comment:
Please use `self.assertRaisesRegex` instead of bare `assertRaises`, like here: https://github.com/python/cpython/blob/dd13c88b5371e13fc16b84e2f9b8715d917de269/Lib/test/test_dataclasses.py#L64-L67

The change gives a clear understanding of what exact error case is tested.

Tests below also should be upgraded

Reply:
Thank you for the review.
I've pushed an update, and will wait for test run (not sure about errno on Windows).
I'll resolve this comment when tests pass (possibly with an extra commit).